src/components/ec-timer/ec-timer.vue

```diff
@@ -74,22 +89,40 @@ const props = defineProps({
 });
 const emit = defineEmits(['time-expired']);
 
-const diameter = computed(() => radius.value * 2 + strokeWidth.value);
+const radius = 24;
+const strokeWidth = 4;
+const diameter = computed(() => radius * 2 + strokeWidth);
 const viewbox = computed(() => `0 0 ${diameter.value} ${diameter.value}`);
-const circumference = computed(() => 2 * Math.PI * radius.value);
+const circumference = computed(() => 2 * Math.PI * radius);
 const steps = computed(() => circumference.value / props.seconds);
-const offset = computed(() => circumference.value + steps.value * secondsLeft.value);
 
-const radius = ref(24);
-const strokeWidth = ref(4);
-const secondsLeft = ref(props.seconds);
+const offset = computed(() => {
+  if (props.isRunning) {
+    return circumference.value + steps.value * totalSecondsLeft.value;
+  }
+  return circumference.value;
+});
+
+const minutesLeft = computed(() => Math.floor(totalSecondsLeft.value / 60));
+
+const secondsLeft = computed(() => {
+  if (!props.isRunning) {
+    return props.seconds;
+  }
+  if (props.showMinutes) {
+    return totalSecondsLeft.value - Math.round(minutesLeft.value * 60);
```

Comment:
```suggestion
    return totalSecondsLeft.value % 60;
```

We could wrap it in `Math.round()` for additional safety but if it's guaranteed that totalSecondsLeft is always int we should safe.